por/minercache.go

```diff
@@ -0,0 +1,56 @@
+package por
+
+import (
+	"sync"
+	"time"
+
+	"github.com/nknorg/nkn/v2/common"
+	"github.com/nknorg/nkn/v2/config"
+)
+
+var (
+	recentMinerLock  sync.Mutex
+	recentMinerCache common.Cache
+)
+
+type RecentMiner map[string]int
+
+func init() {
+	recentMinerCache = common.NewGoCache(5*time.Minute, time.Minute)
+}
+
+func GetRecentMiner(blockHash []byte) (RecentMiner, error) {
+	if v, ok := recentMinerCache.Get(blockHash); ok {
+		if rm, ok := v.(RecentMiner); ok {
+			return rm, nil
+		}
+	}
+
+	recentMinerLock.Lock()
+	defer recentMinerLock.Unlock()
+
+	if v, ok := recentMinerCache.Get(blockHash); ok {
+		if rm, ok := v.(RecentMiner); ok {
+			return rm, nil
+		}
+	}
```

Comment:
Why recentMinerCache.Get twice ?

Reply:
To prevent the cache from being calculated multiple times.